### PR TITLE
Rating price slider filter

### PR DESCRIPTION
**Pull request by Sudhanshusoni45**
### Changes to check for:
- Implemented Price Slider Filter
- Implemented Rating Filter
- exported utils from index.js

### CODE REVIEW

File: src/components/Card/Card.js [hunk 1]
```diff
@@ -33,12 +50,10 @@ const Card = ({ id, name, price, bgColor, inWishList, setProducts }) => {
...
           <h4>${price}</h4>
-
           <div>
-            <i className=" fas fa-star fa-sm"></i>
-            <i className=" fas fa-star fa-sm"></i>
-            <i className=" fas fa-star fa-sm"></i>
-            <i className=" fas fa-star fa-sm"></i>
+            {starArr.map((item) => (
+              <i className=" fas fa-star fa-sm"></i>
+            ))}
```
Comment (kart-c) [inline src/components/Card/Card.js:56]: provide a key attribute

File: src/components/FilterSideBar/FilterSideBar.js [hunk 2]
```diff
@@ -87,7 +95,16 @@ const FilterSideBar = () => {
...
           <h3>Price</h3>
-          <input type="range" />
+          <label htmlFor="priceSlider">$0 to ${rangeValue}</label>
+          <input
+            type="range"
+            min="0"
+            max="1000"
+            value={rangeValue}
+            onChange={(e) =>
+              filterDispatch({ type: "PRICE_FILTER", payload: e.target.value })
```
Comment (kart-c) [inline src/components/FilterSideBar/FilterSideBar.js:111]: use `htmlFor` for label and `id` for input elements for all the labels and inputs. This helps in accessibility

File: src/reducer/filterReducer.js [hunk 3]
```diff
@@ -21,6 +21,21 @@ const filterReducer = (state, action) => {
...
+      return { ...state, rating: "4_STAR_&_ABOVE" };
+
+    case "3_STAR_&_ABOVE":
+      return { ...state, rating: "3_STAR_&_ABOVE" };
+
+    case "2_STAR_&_ABOVE":
+      return { ...state, rating: "2_STAR_&_ABOVE" };
+
+    case "1_STAR_&_ABOVE":
+      return { ...state, rating: "1_STAR_&_ABOVE" };
```
Comment (kart-c) [inline src/reducer/filterReducer.js]: you can optimize this rating filter into a single function by using the `value` attributes of the radio button.

**Review by kart-c — APPROVED** [review]
I have given some suggestions, I would suggest you add the attributes for `label` and `input` elements. It will also help in reviewing the UI fast.

**Review by mohammed-sas — APPROVED** [review]
1.functionality is working as intended
2.filter context and reducer has been coded
3.great ui/ux

**Pull request merged**